include/verilated_random.cpp

```diff
@@ -467,7 +468,17 @@ bool VlRandomizer::parseSolution(std::iostream& f) {
                                 "hex_index contains invalid format");
                     continue;
                 }
-                const long long index = std::stoll(hex_index.substr(start + 2), nullptr, 16);
+                std::string trimmed_hex = hex_index.substr(start + 2);
+                std::string index;
+                if (trimmed_hex.size() <= 8) {  // Small numbers: <= 32 bits
+                    index = std::to_string(
+                        std::stoll(trimmed_hex, nullptr, 16));  // Convert to decimal
```

Comment:
```suggestion
                    oss << "[" << .... << "]"
```
Avoids making temp index string, then do similar on else branch.